ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml

```diff
@@ -0,0 +1,29 @@
+import QtQuick 2.13
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+StatusBaseText {
+    id: root
+    property bool loading: false
+    property color customColor: Theme.palette.directColor1
+    color: loading ? "transparent" : customColor
+    Loader {
+        anchors.left: parent.left
+        anchors.verticalCenter: parent.verticalCenter
+        active: root.loading
+        sourceComponent: LoadingComponent {
+            anchors.centerIn: parent
+            radius: textMetrics.font.pixelSize === 15 ? 4 : 8
+            height: textMetrics.tightBoundingRect.height
+            width: Math.min(root.width, 140)
+            loading: root.loading
+        }
+    }
+    TextMetrics {
```

Comment:
is this used somewhere?

Reply:
yes I need it for getting the height of the letter